src/index.test.ts

```diff
@@ -3714,40 +3714,78 @@ describe('normalize', function () {
-  const recovered = sigUtil.recoverPersonalSignature(
-    msgParams as sigUtil.SignedMsgParams<string>,
+describe('personalSign', function () {
+  const privateKey = Buffer.from(
+    '4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0',
+    'hex',
   );
+  const helloWorldSignature =
+    '0x90a938f7457df6e8f741264c32697fc52f9a8f867c52dd70713d9d2d472f2e415d9c94148991bbe1f4a1818d1dff09165782749c877f5cf1eff4ef126e55714d1c';
+  const helloWorldMessage = 'Hello, world!';
 
-  expect(recovered).toBe(address);
-});
+  it('should sign a message', function () {
+    expect(sigUtil.personalSign(privateKey, { data: helloWorldMessage })).toBe(
+      helloWorldSignature,
+    );
+  });
 
-it('personalSign and extractPublicKey', function () {
-  const privKeyHex =
-    '4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0';
-  const pubKeyHex =
-    '0x9e9e45b2ec5f070b4e26f57c7fedf647afa7a03e894789816fbd12fedc5acd79d0dfeea925688e177caccb8f5e09f0c289bbcfc7adb98d76f5f8c5259478903a';
+  it('should recover the address from a signature', function () {
+    const address = ethUtil.addHexPrefix(
+      ethUtil.privateToAddress(privateKey).toString('hex'),
+    );
 
-  const privKey = Buffer.from(privKeyHex, 'hex');
-  const message = 'Hello, world!';
-  const msgParams: sigUtil.MsgParams<string> = { data: message };
+    expect(
+      sigUtil.recoverPersonalSignature({
+        data: helloWorldMessage,
+        sig: helloWorldSignature,
+      }),
+    ).toBe(address);
+  });
 
-  const signed = sigUtil.personalSign(privKey, msgParams);
-  msgParams.sig = signed;
-  const publicKey = sigUtil.extractPublicKey(
-    msgParams as sigUtil.SignedMsgParams<string>,
-  );
+  it('should recover the public key from a signature', function () {
+    const publicKey = ethUtil.addHexPrefix(
+      ethUtil.privateToPublic(privateKey).toString('hex'),
+    );
+
+    expect(
+      sigUtil.extractPublicKey({
+        data: helloWorldMessage,
+        sig: helloWorldSignature,
+      }),
+    ).toBe(publicKey);
+  });
 
-  expect(publicKey).toBe(pubKeyHex);
+  it('should sign a message and recover the address of the signer', function () {
+    const address = ethUtil.addHexPrefix(
+      ethUtil.privateToAddress(privateKey).toString('hex'),
+    );
+    const signature = sigUtil.personalSign(privateKey, {
+      data: helloWorldMessage,
+    });
+
+    expect(
+      sigUtil.recoverPersonalSignature({
+        data: helloWorldMessage,
+        sig: signature,
+      }),
+    ).toBe(address);
+  });
+
+  it('should sign a message and recover the public key of the signer', function () {
+    const publicKey = ethUtil.addHexPrefix(
+      ethUtil.privateToPublic(privateKey).toString('hex'),
+    );
+    const signature = sigUtil.personalSign(privateKey, {
+      data: helloWorldMessage,
+    });
+
+    expect(
+      sigUtil.extractPublicKey({
+        data: helloWorldMessage,
+        sig: signature,
+      }),
+    ).toBe(publicKey);
+  });
```

Comment:
The purpose of these tests appears to be to convince ourselves that we can create new, valid signatures and extract the public key and address from them. That being the case, wouldn't it be better to use some other data to sign that the "hello world" message? Otherwise I don't think they're telling us anything we don't already now.

Reply:
We talked this over, it's fine.

Reply:
The other two tests ensure that we generate the same signature we did before, and that we can recover the same signature we did before. These are important because signatures might be stored by third parties, so any change to the format would be breaking.

But this test just ensures the public key is recoverable, irrespective of whether the signature has changed or not. This test would be useful if we ever intentionally changed the signature output for some reason. Or I guess it might be useful in pinpointing what went wrong if we accidentally changed it.